infra/src/main/kotlin/fi/fta/geoviite/infra/ratko/RatkoOperatingPointDao.kt

```diff
@@ -107,15 +120,37 @@ class RatkoOperatingPointDao(jdbcTemplateParam: NamedParameterJdbcTemplate?) : D
                 "layout_srid" to LAYOUT_SRID.code
             )
         ) { rs, _ ->
-            RatkoOperatingPoint(
-                rs.getOid("external_id"),
-                rs.getString("name"),
-                rs.getString("abbreviation"),
-                rs.getString("uic_code"),
-                rs.getEnum("type"),
-                rs.getPoint("x", "y"),
-                rs.getIntId("track_number_id"),
+            toRatkoOperatingPoint(rs)
+        }
+    }
+
+    @Transactional(readOnly = true)
+    fun searchOperatingPoints(searchTerm: FreeText, resultLimit: Int): List<RatkoOperatingPoint> {
+        val sql = """
+            select
+              external_id,
+              name,
+              abbreviation,
+              uic_code,
+              type,
+              postgis.st_x(location) as x,
+              postgis.st_y(location) as y,
+              track_number_id
+              from layout.operating_point
+              where name ilike :searchPattern
+              or abbreviation ilike :searchPattern
```

Comment:
idea ei osaa automaattisisentää näitä monirivisiä sql:iä, mutta sen voisi silti tehdä. Ainakin omaan silmään on paljon helpompi lukea kun nuo pääsanat on ylempänä ja niille alisteiset jutut selvästi niiden alla. Yksi tapa on heittää koko roska konsoliin, automaati-indentoida siellä ja kopsia takaisin. Tai sitten vain käsin.